shell/common/node_bindings.cc

```diff
@@ -476,25 +475,33 @@ node::Environment* NodeBindings::CreateEnvironment(
     // not to register its handler (overriding blinks) in non-browser processes.
     flags |= node::EnvironmentFlags::kNoRegisterESMLoader |
              node::EnvironmentFlags::kNoInitializeInspector;
-    v8::TryCatch try_catch(context->GetIsolate());
-    env = node::CreateEnvironment(
-        isolate_data_, context, args, exec_args,
-        static_cast<node::EnvironmentFlags::Flags>(flags));
-    DCHECK(env);
-
-    // This will only be caught when something has gone terrible wrong as all
-    // electron scripts are wrapped in a try {} catch {} by webpack
-    if (try_catch.HasCaught()) {
-      LOG(ERROR) << "Failed to initialize node environment in process: "
-                 << process_type;
+  }
+
+  v8::TryCatch try_catch(isolate);
+  env = node::CreateEnvironment(
+      isolate_data_, context, args, exec_args,
+      static_cast<node::EnvironmentFlags::Flags>(flags));
+
+  if (try_catch.HasCaught()) {
+    std::string err_msg =
+        "Failed to initialize node environment in process: " + process_type;
+    v8::Local<v8::Value> exception = try_catch.Exception();
+    if (!exception->IsNull() && exception->IsObject()) {
+      v8::MaybeLocal<v8::Value> maybe_message = exception.As<v8::Object>()->Get(
+          context, gin::ConvertToV8(isolate, "message"));
+      v8::Local<v8::Value> error_message;
+      if (maybe_message.ToLocal(&error_message) && error_message->IsString()) {
+        std::string msg;
+        if (gin::ConvertFromV8(isolate, error_message, &msg))
+          err_msg += " with error: " + msg;
+      }
```

Comment:
```suggestion
   v8::Local<v8::Message> message = try_catch.Message();
   std::string msg;
   if (!message.IsEmpty() && gin::ConvertFromV8(isolate, message->Get(), &msg))
       err_msg += " , with error: " + msg;
```

Reply:
@deepak1556 i went back to this and i do specifically want the message body - `message->Get()` returns the whole of `Uncaught Error: error` when i only want `error`, which is why i went with the previous approach. Are you ok with this or would you prefer to keep the `Uncaught Error` portion?

Reply:
I think the retrieving the whole message is fine, also saves extra checks and conversions to v8 in the previous approach, whereas the current version gets the object already available in c++ and only requires a single conversion from v8.